mllib/src/main/scala/org/apache/spark/mllib/feature/HashingTF.scala

```diff
@@ -53,9 +65,10 @@ class HashingTF(val numFeatures: Int) extends Serializable {
   @Since("1.1.0")
   def transform(document: Iterable[_]): Vector = {
     val termFrequencies = mutable.HashMap.empty[Int, Double]
+    val setTF = if (binary) (i: Int) => 1.0 else (i: Int) => termFrequencies.getOrElse(i, 0.0) + 1.0
```

Comment:
Are you guys talking about the `document.foreach` loop?  The if-then-else is just to assign a function, so it would just be called once per document, then the function is called in the loop.  If this line is a bit too much, I could rewrite it to something more straightforward.